narayana/ArjunaJTA/jta/tests/classes/org/jboss/narayana/rts/EmptyTxnTest.java

```diff
@@ -50,15 +49,19 @@ public static void tearDown() throws IOException {
 
     @Benchmark
     public void testEmptyTxn(Blackhole bh) throws IOException {
-        bh.consume(beginTx());
+        Set<Link> links = beginTx();
         // make two service calls for comparison with RTSTests#testTxn
         bh.consume(sendRequest());
         bh.consume(TxnHelper.endTxn(txnClient, links));
     }
 
-    public boolean beginTx() throws IOException {
-        links = TxnHelper.beginTxn(txnClient, TXN_URL);
-        return true;
+    @Test
+    public void testEmptyTxn() throws IOException {
+        testEmptyTxn(new Blackhole("Today's password is swordfish. I understand instantiating Blackholes directly is dangerous."));
```

Comment:
In fact the authors of JMH frown on it:

```
Either we instantiate from the JMH generated code,
            // or our user is a tricky bastard, and gets what's coming to him.
```

Reply:
Nice finding ;) 
I wanted to test the testEmptyTxn method both with junit and jmh. For jmh the Blackhole has been added in the PR related to this issue https://issues.redhat.com/browse/JBTM-3137. However for junit test it is not allowed to have arguments in the method declarations. I can therefore remove junit methods or duplicate the benchmark tests and test them with junit removing the Blackhole  arguments. WDYT?

Reply:
It's certainly nice to have junit tests alongside the benchmark to facilitate debugging but if it means doing something the JMH developers did not intend then I think the cons outweigh the pros. I would vote to remove them but let's pull in @jmfinelli to get his opinion as well.

Reply:
I see your point Michael so I implemented the Junit tests removing the forced instantiation of Blackhole. Would you mind have a look at the new implementation now? Thanks again.

Reply:
I don't know anything about JMH and benchmarking. Having said that...according to JMH's javadoc:

**Prevent instantiation by user code. Without additional countermeasures to properly escape Blackhole, its magic is not working**. The instances of Blackholes which are injected into benchmark methods are treated by JMH,
and **users are supposed to only use the injected instances**.

It only *seems* simple to make the constructor non-public, but then there is a lot of infrastructure code which assumes @State has a default constructor. One might suggest doing the internal factory method to instantiate,
but that does not help when extending the Blackhole. There is a *messy* way to special-case most of these problems within the JMH code, but it does not seem to worth the effort.

**Therefore, we choose to fail at runtime. It will only affect the users who thought "new Blackhole()" is a good idea, and _these users are rare. If you are reading this comment, you might be one of those users. Stay cool! Don't instantiate Blackholes directly though._**

### My 2c
The design choice of JMH's developers is clearly explained in the javadoc ("Stay cool! Don't instantiate Blackholes directly though"). IMO, it isn't a great idea to going against this.

Reply:
@jmfinelli I tagged you to get input on removing the junit tests from the module since the intention is to test performance not functionality, sorry if I wasn't clear about that.

Reply:
> Personally I think the unit tests should be removed because it has resulted in this misuse of JMH.

I agree with you

Reply:
> I see your point Michael so I implemented the Junit tests removing the forced instantiation of Blackhole. Would you mind having a look at the new implementation now? Thanks again.

Thanks @marcosgopen

Reply:
@marcosgopen how do you wish to proceed?

Reply:
Since the task is not related to Junit tests I am ok with removing those Junit tests I added.

Reply:
In that case I will resolve this conversation.